### PR TITLE
Consolidation of the priority problem statement

### PR DESCRIPTION
**Pull request by LPardue**
On the priority team mailing list I advocated for a [consolidation of the problem statement](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/http-priorities-design-team/an06MqgKgIE)

This PR builds on top of #58, creating a new section that combines that text, with text from the introduction and pulls in some of Roberto's text. All of that happens on https://github.com/kazuho/draft-kazuho-httpbis-priority/commit/b2b927ccbeea76f9edcb5e49ad9be06530d9e2ab

@rmarx I didn't pull anything specifically from your paper but if you feel there is something that I don't cover then please do suggest it

### CODE REVIEW

File: draft-kazuho-httpbis-priority.md [hunk 1]
```diff
@@ -98,6 +80,117 @@ The terms sh-token and sh-boolean are imported from
...
+an important realization in the design of HTTP/2. Prioritization is a
+difficult problem, so it will always be suboptimal, particularly if one endpoint
+operates in ignorance of the needs of its peer.
+
+HTTP/2 introduced a complex prioritization signaling scheme that used a
+combination of dependencies and weights, formed into an unbalanced tree. This
+scheme suffers from poor deployment and interoperability.
+
+The rich flexibility permitted in tree building is rarely excersised by clients;
+they generally either chosen a model optimized for a web use case or do nothing
```
Comment (kazuho) [inline draft-kazuho-httpbis-priority.md]: Maybe there is an error in tense? Something like "they generally either have chosen .. ore have done nothing" might flow better.

File: draft-kazuho-httpbis-priority.md [hunk 2]
```diff
@@ -98,6 +80,117 @@ The terms sh-token and sh-boolean are imported from
...
+want to prioritize the delivery of images that are critical to user experience
+above other images, but below the CSS files. But with the HTTP/2 prioritization
+scheme, it is impossible for the server to determine how such images should be
+prioritized against other responses that use the client-driven prioritization
+tree, because every client builds the HTTP/2 prioritization tree in a different
+way.
+
+Intermediaries can coalesce multiple HTTP/2 clients into a single upstream
+HTTP/2 connection and the prioritization scheme allows for this. However most
+intermediaries do not support this and it does not define a method that can be
```
Comment (kazuho) [inline draft-kazuho-httpbis-priority.md]: Maybe at least split the sentence, as the topic seems to change? The first half of this paragraph (up to "do not support this") talks about passing the tree through intermediaries, while the second half discusses about the incapability of the prioritization scheme not allowing the server to express their priorities.

**Review by kazuho — COMMENTED** [review]
Thank you for the PR. The problem statement looks good modulo the editorial points below.

File: draft-kazuho-httpbis-priority.md (hunk 2 above)
Comment (LPardue) [reply draft-kazuho-httpbis-priority.md]: oops I messed this up a bit, PTAL at https://github.com/kazuho/draft-kazuho-httpbis-priority/pull/61/commits/7cfe822c09cfd9457d0acc95a1696d58a7cc1628

File: draft-kazuho-httpbis-priority.md (hunk 1 above)
Comment (LPardue) [reply draft-kazuho-httpbis-priority.md]: On a reread, I noticed an opportunity to make this paragraph stonger by stealing text from the next one. PTAL at https://github.com/kazuho/draft-kazuho-httpbis-priority/pull/61/commits/7cfe822c09cfd9457d0acc95a1696d58a7cc1628

File: draft-kazuho-httpbis-priority.md [hunk 3]
```diff
@@ -92,26 +92,27 @@ HTTP/2 introduced a complex prioritization signaling scheme that used a
...
-tree, because every client builds the HTTP/2 prioritization tree in a different
-way.
-
-Intermediaries can coalesce multiple HTTP/2 clients into a single upstream
-HTTP/2 connection and the prioritization scheme allows for this. However most
-intermediaries do not support this and it does not define a method that can be
-used by a server to express the priority of a response. Without such a method,
-intermediaries cannot coordinate client-driven and server-driven priorities.
+The rich flexibility of client-driven HTTP/2 prioritization tree building is
+rarely excercised; experience shows that clients either choose a single model
```
Comment (kazuho) [inline draft-kazuho-httpbis-priority.md]: s/excercised/exercised/

File: draft-kazuho-httpbis-priority.md [hunk 4]
```diff
@@ -92,26 +92,27 @@ HTTP/2 introduced a complex prioritization signaling scheme that used a
...
+optimized for a web use case (and don't vary it) or do nothing at all. But every
+client builds their in a different way, which makes it difficult for servers to
```
Comment (kazuho) [inline draft-kazuho-httpbis-priority.md]: s/their/theirs/ or "their prioritization trees"

File: draft-kazuho-httpbis-priority.md [hunk 5]
```diff
@@ -92,26 +92,27 @@ HTTP/2 introduced a complex prioritization signaling scheme that used a
...
+understand their intent and act accordingly.
+
+Many server implementations do not include support for the HTTP/2 priority
```
Comment (kazuho) [inline draft-kazuho-httpbis-priority.md]: Maybe it's better to say "Many HTTP/2 server implementations"?

**Review by kazuho — COMMENTED** [review]
Thank you for the changes. LGTM modulo minor editorial suggestions.

**Review by kazuho — APPROVED** [review]

**Comment by rmarx** [comment]
Hello both,

Sorry for the late reply, have something else I need to finish today.

This PR looks good overall, I would just have included a section indicating that research and experiments have shown that simpler schemes can achieve similar performance as the more complex, dependency-tree driven schemes we've seen in the wild for HTTP/2. And that, as such, this initial document focuses on a simple and straightforward scheme as a result. If you agree, I can make a new PR for that, since this one was just merged :)

**Comment by kazuho** [comment]
@rmarx That sounds like something highly desirable to clarify. I think that a PR would be more than welcome.

**Pull request merged**